### PR TITLE
test(lint): harden the getInstance leak guard against import aliases, cross-symbol captures, and unrelated spy restores

### PR DESCRIPTION
**Pull request by kaeawc**
Follow-up to [#7058](https://github.com/kaeawc/auto-mobile/pull/7058) (issue [#7052](https://github.com/kaeawc/auto-mobile/issues/7052)). The merged guard classified `<Symbol>.getInstance` install-vs-restore by literal identifier spelling, a name-only capture set, and a file-wide `mockRestore` acquittal. A codex review of that PR (after it had already merged) identified three residual false negatives; this closes all three:

- **Import aliases**: `symbolOf` now resolves local identifiers through a map built from the file's named-import specifiers (`propertyName` to alias) before falling back to literal spelling, so `import { AndroidCtrlProxyClient as Client }` + `Client.getInstance = …` is inventoried and checked.
- **Cross-symbol capture**: captured-original tracking is a scope-aware `binding -> CtrlProxySymbol` map, and a reassignment counts as a restore only when the captured symbol matches the assignment target, so a copy/paste bug that captures Android's original but assigns it onto iOS is flagged as a leak.
- **Spy-restore over-acquittal**: installs are split into direct-assignment vs spy-handle installs, and each kind is gated on its own matching restore mechanism, so an unrelated `spyOn(...).mockRestore()` no longer acquits a separate un-restored direct assignment.

Three new self-tests (alias install without restore; cross-symbol mismatch; spy-restore-does-not-acquit-direct-install), each failing-first then green. The real tree was re-audited: no production test file needed a teardown change. `bun test test/daemon test/server` (the #7052 combined repro) stays 0 fail.

These arrived as review threads on #7058 after it merged, so they land here rather than on the merged branch.

### CODE REVIEW

File: test/lint/ctrlProxyGetInstanceRestore.test.ts [hunk 1]
```diff
@@ -104,19 +108,28 @@ describe("CtrlProxy getInstance mocks are restored in-file (issue #7052)", () =>
...
+  function symbolOf(
+    node: ts.Expression,
+    symbolAliases: ReadonlyMap<string, CtrlProxySymbol>,
+  ): CtrlProxySymbol | undefined {
     const core = unwrap(node);
-    if (ts.isIdentifier(core) && SYMBOL_SET.has(core.text)) {
-      return core.text as CtrlProxySymbol;
+    if (ts.isIdentifier(core)) {
+      return (
+        symbolAliases.get(core.text) ??
```
[the remark this answers is not in the document]
Comment (kaeawc) [reply test/lint/ctrlProxyGetInstanceRestore.test.ts:118]: Valid edge cases, but all three are theoretical (no test file in the tree triggers them; the guard passes on the real tree) and this is the third precision round on a test-only backstop. Per the round-limit policy I'm tracking them in follow-up https://github.com/kaeawc/auto-mobile/issues/7065 rather than adding a 4th round here; noted that threads 2 and 3 are latent false positives worth prioritizing. This PR already strictly improves guard detection over main.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/lint/ctrlProxyGetInstanceRestore.test.ts [hunk 2]
```diff
@@ -197,74 +216,151 @@ describe("CtrlProxy getInstance mocks are restored in-file (issue #7052)", () =>
...
+      let id = scopeIds.get(scope);
+      if (id === undefined) {
+        id = nextScopeId++;
+        scopeIds.set(scope, id);
+      }
+      return `${id}:${name}`;
+    };
+    const declarationKey = (name: ts.Identifier): string => {
+      const [scope] = scopeChain(name);
+      return keyForScope(scope, name.text);
```
[the remark this answers is not in the document]
Comment (kaeawc) [reply test/lint/ctrlProxyGetInstanceRestore.test.ts:258]: Valid edge cases, but all three are theoretical (no test file in the tree triggers them; the guard passes on the real tree) and this is the third precision round on a test-only backstop. Per the round-limit policy I'm tracking them in follow-up https://github.com/kaeawc/auto-mobile/issues/7065 rather than adding a 4th round here; noted that threads 2 and 3 are latent false positives worth prioritizing. This PR already strictly improves guard detection over main.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/lint/ctrlProxyGetInstanceRestore.test.ts [hunk 3]
```diff
@@ -273,20 +369,28 @@ describe("CtrlProxy getInstance mocks are restored in-file (issue #7052)", () =>
...
     const leaks: string[] = [];
-    for (const symbol of facts.installs) {
-      if (!facts.restores.has(symbol) && !facts.restoredBySpy.has(symbol)) {
+    for (const symbol of facts.directInstalls) {
+      if (!facts.restores.has(symbol)) {
+        leaks.push(`${file}: installs ${symbol}.getInstance but never restores it`);
+      }
+    }
+    for (const symbol of facts.spyInstalls) {
+      if (!facts.restoredBySpy.has(symbol)) {
```
[the remark this answers is not in the document]
Comment (kaeawc) [reply test/lint/ctrlProxyGetInstanceRestore.test.ts:393]: Valid edge cases, but all three are theoretical (no test file in the tree triggers them; the guard passes on the real tree) and this is the third precision round on a test-only backstop. Per the round-limit policy I'm tracking them in follow-up https://github.com/kaeawc/auto-mobile/issues/7065 rather than adding a 4th round here; noted that threads 2 and 3 are latent false positives worth prioritizing. This PR already strictly improves guard detection over main.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
